syntax = "proto3";

package pack.color;

message Rgba {
  message AsFloats {
    // Values should be on [0., 1.].
    float red = 1;
    float green = 2;
    float blue = 3;
    float alpha = 4;
  }
  message AsInts {
    // Values should be on [0, 255].
    int32 red = 1;
    int32 green = 2;
    int32 blue = 3;
    int32 alpha = 4;
  }
  message AsUint32s {
    // Values should be on [0, std::numeric_limits<uint32_t>::max()].
    fixed32 red = 1;
    fixed32 green = 2;
    fixed32 blue = 3;
    fixed32 alpha = 4;
  }
  oneof color_representation {
    AsFloats float_values = 1;
    AsInts int_values = 2;
    AsUint32s uint32_values = 3;
  }
}
